query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapQuerySpec.groovy

```diff
@@ -0,0 +1,242 @@
+        def missingBindMethodMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, LdapBindMethod.DEFAULT_FIELD_NAME]
+        def missingQueryBaseMsgPath = baseMsg + [LdapQuery.QUERY_BASE_FIELD_NAME]
+        def missingQueryMsgPath = baseMsg + [LdapQueryField.DEFAULT_FIELD_NAME]
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 8
+        report.messages().count {
+            it.getCode() == DefaultMessages.MISSING_REQUIRED_FIELD
+        } == 8
+
+        report.messages()*.getPath() as Set == [missingHostMsgPath, missingPortMsgPath, missingEncryptMsgPath,
+                                                missingUsernameMsgPath, missingUserpasswordMsgPath, missingBindMethodMsgPath,
+                                                missingQueryBaseMsgPath, missingQueryMsgPath] as Set
+    }
+
+    def 'Fail to connect to LDAP'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().port(666).getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().getValue(),
+                (LdapQuery.QUERY_BASE_FIELD_NAME)       : exampleQueryBase().getValue(),
+                (LdapQueryField.DEFAULT_FIELD_NAME)     : exampleLdapQuery().getValue()]
+        ldapQueryFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).getCode() == DefaultMessages.CANNOT_CONNECT
+        report.messages().get(0).getPath() == [LdapQuery.FIELD_NAME, BaseFunctionField.ARGUMENT, LdapConnectionField.DEFAULT_FIELD_NAME]
+
+    }
+
+    def 'Fail to bind user over bind method "Simple" with bad password'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().password("badPassword").getValue(),
+                (LdapQuery.QUERY_BASE_FIELD_NAME)       : exampleQueryBase().getValue(),
+                (LdapQueryField.DEFAULT_FIELD_NAME)     : exampleLdapQuery().getValue()]
+
+        ldapQueryFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).getCode() == LdapMessages.CANNOT_BIND
+        report.messages().get(0).getPath() == [LdapQuery.FIELD_NAME, BaseFunctionField.ARGUMENT, LdapBindUserInfo.DEFAULT_FIELD_NAME]
+    }
+
+    def 'Successfully query with entries found (No encryption)'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().password("secret").getValue(),
```

Comment:
It's not retrieving the password, it's setting it then turning bindInfo into a map.